src/dune/dune_file.ml

```diff
@@ -1916,8 +1917,10 @@ module Toplevel = struct
        and+ variants = variants_field
        and+ libraries =
          field "libraries" (repeat (located Lib_name.decode)) ~default:[]
+       and+ pps =
+         field "preprocess" Preprocess.decode ~default:Preprocess.No_preprocessing
```

Comment:
This field should require 2.5 as the minimum version of dune.

Reply:
By the way, I think it's better to add the error checking for `future_syntax` and `action` right after the parsing. This way, we can confidently assume that no rules will ever encounter the unsupported configuration and can safely `assert false` on those cases.